From 72aaa3d28daa1b187059ef148c45bd9f186b430d Mon Sep 17 00:00:00 2001
From: Matti Kosola <matti.kosola@jolla.com>
Date: Thu, 9 Jul 2020 11:22:10 +0000
Subject: [PATCH 2/4] Set proper llvm targets.

Signed-off-by: Matti Kosola <matti.kosola@jolla.com>
---
 config.toml.example | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/config.toml.example b/config.toml.example
index 9121d1e1799..7990f01ab42 100644
--- a/config.toml.example
+++ b/config.toml.example
@@ -63,7 +63,7 @@
 # support. You'll need to write a target specification at least, and most
 # likely, teach rustc about the C ABI of the target. Get in touch with the
 # Rust team and file an issue if you need assistance in porting!
-#targets = "AArch64;ARM;Hexagon;MSP430;Mips;NVPTX;PowerPC;RISCV;Sparc;SystemZ;WebAssembly;X86"
+targets = "X86;ARM;AArch64"
 
 # LLVM experimental targets to build support for. These targets are specified in
 # the same format as above, but since these targets are experimental, they are
-- 
2.20.1

